RFS/src/main/java/org/opensearch/migrations/metadata/CreationResult.java

```diff
@@ -0,0 +1,46 @@
+package org.opensearch.migrations.metadata;
+
+import java.util.Optional;
+
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Data;
+import lombok.Getter;
+
+@Builder
+@Data
+public class CreationResult implements Comparable<CreationResult>  {
+    private final String name;
+    private final Exception exception;
+    private final CreationFailureType failureType;
+    public boolean wasSuccessful() {
+        return getFailureType() == null; 
+    }
+
+    public boolean wasFatal() {
+        return Optional.ofNullable(getFailureType()).map(CreationFailureType::isFatal)
+            .orElse(false);
+    }
+
+    @AllArgsConstructor
+    @Getter
+    public static enum CreationFailureType {
+        ALREADY_EXISTS(false, "already exists"),
+        DEPENDENCY_FAILURE(false, "dependency failure"),
```

Comment:
Can you describe DEPENDENCY_FAILURE and why it's not fatal?

Reply:
I've removed this kind of failure since it was not as well defined as we would like.